### PR TITLE
Support for locals

### PR DESCRIPTION
**Pull request by peters**
A nice feature inspired by the express framework on how local variables should be exposed to views and other internal components. This component provides a simple way of doing so.

```

<?php
require 'Slim/Slim.php';
$app = new Slim();
$app->locals('name', function($lastname) {
   return "Peter $lastname";
});

$app->get('/hello/:name', function ($name) use($app) {
    echo "Hello, " . $app->locals('name', "Sunde"));
});
$app->run();
```

### CODE REVIEW

**Comment by peters** [comment]
Needs improvement, but it's a start :) 

Should i push to the develop branch instead?

File: Slim/Locals.php [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+		return $value;
+	}
+	
+	/**
+	 * Prevent properties being added to class object
+	 * 
+	 * @param string $name 
+	 * @param mixed $value
+	 */
+	public function __set($name, $value) {}
```
Comment (Baachi) [inline Slim/Locals.php:47]: Can be removed?

File: Slim/Locals.php [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+	/**
+	 * @var array
+	 */
+	private $properties;
+
+ 	/**
+     * Constructor
+     *
+     *
```
Comment (Baachi) [inline Slim/Locals.php:13]: Useless empty lines here.

**Comment by codeguy** [comment]
Please move major changes into separate pull requests. For example, one PR for locals, one PR for auth, one PR for travis. Unusable as is.

**Comment by peters** [comment]
Sorry about the late feedback, i will provide proper pull requests tonight or tomorrow :)

**Pull request closed without merging**